lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -593,9 +599,9 @@ public void OnNext(IFailedContext failedContext)
         {
             lock (_lock)
             {
-                if (_taskManager.AreAllTasksCompleted())
+                if (_systemState.CurrentState.Equals(SystemState.TasksCompleted))
```

Comment:
This would probably look better as a `switch (_systemState.CurrentState)`, similar to other handlers. We should probably ignore this in `Fail`, `ShuttingDown` and `TasksCompleted` and throw exception in all other states